misc/toml-parser/src/main/java/org/ballerinalang/toml/parser/ManifestBuildListener.java

```diff
@@ -0,0 +1,258 @@
+import org.ballerinalang.toml.model.fields.DependencyField;
+import org.ballerinalang.toml.model.fields.ManifestHeader;
+import org.ballerinalang.toml.model.fields.PackageField;
+import org.ballerinalang.toml.util.SingletonStack;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Custom listener which is extended from the Toml listener with our own custom logic.
+ *
+ * @since 0.964
+ */
+public class ManifestBuildListener extends TomlBaseListener {
+    private final Manifest manifest;
+    private final SingletonStack<String> currentKey = new SingletonStack<>();
+    private Dependency dependency;
+    private String currentHeader = null;
+
+    /**
+     * Cosntructor with the manifest object.
+     *
+     * @param manifest manifest object
+     */
+    ManifestBuildListener(Manifest manifest) {
+        this.manifest = manifest;
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * <p>The default implementation does nothing.</p>
+     *
+     * @param ctx
+     */
+    @Override
+    public void enterKeyval(TomlParser.KeyvalContext ctx) {
+        currentKey.push(ctx.key().getText());
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * <p>The default implementation does nothing.</p>
+     *
+     * @param ctx
+     */
+    @Override
+    public void enterString(TomlParser.StringContext ctx) {
+        setToManifest(ctx.getText());
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * <p>The default implementation does nothing.</p>
+     *
+     * @param ctx
+     */
+    @Override
+    public void exitKeyval(TomlParser.KeyvalContext ctx) {
+        setDependancyAndPatches();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * <p>The default implementation does nothing.</p>
+     *
+     * @param ctx
+     */
+    @Override
+    public void enterArray(TomlParser.ArrayContext ctx) {
+        setToManifest(ctx.arrayValues());
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * <p>The default implementation does nothing.</p>
+     *
+     * @param ctx
+     */
+    @Override
+    public void enterStdTable(TomlParser.StdTableContext ctx) {
+        addHeader(ctx.key().getText());
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * <p>The default implementation does nothing.</p>
+     *
+     * @param ctx
+     */
+    @Override
+    public void enterInlineTable(TomlParser.InlineTableContext ctx) {
+        setToManifest(ctx.inlineTableKeyvals());
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * <p>The default implementation does ncommentStartSymbol : HASH;
+     * nonEol :  '\r'| '\t';
+     * <p>
+     * comment :  commentStartSymbol wsCommentNewline basicChar* nonEol*;othing.</p>
+     *
+     * @param ctx
+     */
+    @Override
+    public void exitInlineTable(TomlParser.InlineTableContext ctx) {
+        setDependancyAndPatches();
+    }
+
+    /**
+     * Add the dependencies and patches to the manifest object.
+     */
+    private void setDependancyAndPatches() {
```

Comment:
Typo? Need to be setDependencyAndPatches

Reply:
Removed all the custom listeners from the toml-parser